src/operator/control_flow.cc

```diff
@@ -363,6 +363,393 @@ ForeachGradient(const nnvm::NodePtr& n, const std::vector<nnvm::NodeEntry>& ogra
   return entries;
 }
 
+struct WhileLoopParam : public dmlc::Parameter<WhileLoopParam> {
+  int num_args;
+  int num_outputs;
+  int num_out_data;
+  int max_iterations;
+  nnvm::Tuple<dim_t> cond_input_locs;
+  nnvm::Tuple<dim_t> cond_var_locs;
```

Comment:
please comment the difference between these two fields.

Reply:
will do